### PR TITLE
Fix schedule order via interval-rewrite rules

### PR DESCRIPTION
**Pull request by yihozhang**
Alternative to #680

### CODE REVIEW

File: dag_in_context/src/optimizations/ivt.egg [hunk 1]
```diff
@@ -123,6 +123,6 @@
     (union new-expr loop)
     (union new-loop-context (InLoop (Arg argI new-if-true-ctx) new-loop-outputs))
 
-    (delete (DoWhile inpW outW))
+    (subsume (DoWhile inpW outW))
```
Comment (oflatt) [inline dag_in_context/src/optimizations/ivt.egg:126]: Is this an unrelated change?
I had a weird bug with this this morning that we should investigate where nothing could be extracted

**Review by oflatt — CHANGES_REQUESTED** [review]
Yay!

File: dag_in_context/src/optimizations/ivt.egg (hunk 1 above)
Comment (yihozhang) [reply dag_in_context/src/optimizations/ivt.egg:126]: I also had this in my previous PR. It just makes our rules more consistent (always use subsume rather than delete). 

Your bug might be caused by the wrong schedule order that the other changes in this PR fix? I had a similar issue and that was actually the original motivation for this PR.

**Review by oflatt — APPROVED** [review]
Looks good!
